cl_manycore/libraries/bsg_manycore_cuda.cpp

```diff
@@ -1365,3 +1258,599 @@ static hb_mc_idx_t hb_mc_get_tile_id (hb_mc_coordinate_t origin, hb_mc_dimension
+					__func__,
+					hb_mc_coordinate_get_x(tiles[tile_id]),
+					hb_mc_coordinate_get_y(tiles[tile_id]));
+			return error;
+		}
+
+
+		error = hb_mc_tile_set_coord_symbols(	device->mc, map,
+							device->program->bin, device->program->bin_size,
+							&(tiles[tile_id]), &coord);
+		if (error != HB_MC_SUCCESS) { 
+			bsg_pr_err(	"%s: failed to set tile (%d,%d) coordinate symbols __bsg_x/y.\n",
+					__func__,
+					hb_mc_coordinate_get_x(tiles[tile_id]),
+					hb_mc_coordinate_get_y(tiles[tile_id]));
+			return error;
+		}
+
+
+		error = hb_mc_tile_set_id_symbol(	device->mc, map,
+							device->program->bin, device->program->bin_size,
+							&(tiles[tile_id]),
+							&(coord), &(tg_dim));
+		if (error != HB_MC_SUCCESS) { 
+			bsg_pr_err(	"%s: failed to set tile (%d,%d) id symbol __bsg_id.\n",
+					__func__,
+					hb_mc_coordinate_get_x(tiles[tile_id]),
+					hb_mc_coordinate_get_y(tiles[tile_id]));
+			return error;
+		}
+
+
+		error = hb_mc_tile_set_tile_group_id_symbols(	device->mc, map,
+								device->program->bin, device->program->bin_size,
+								&(tiles[tile_id]), &(tg_id));
+		if (error != HB_MC_SUCCESS) { 
+			bsg_pr_err(	"%s: failed to set tile (%d,%d) tile group id symbold __bsg_tile_group_id.\n",
+					__func__,
+					hb_mc_coordinate_get_x(tiles[tile_id]),
+					hb_mc_coordinate_get_y(tiles[tile_id]));
+			return error;
+		}
+
+
+		error = hb_mc_tile_set_grid_dim_symbols(device->mc, map,
+							device->program->bin, device->program->bin_size,
+							&(tiles[tile_id]), &(grid_dim));
+		if (error != HB_MC_SUCCESS) { 
+			bsg_pr_err(	"%s: failed to set tile (%d,%d) grid size symbol __bsg_grid_size.\n",
+					__func__,
+					hb_mc_coordinate_get_x(tiles[tile_id]),
+					hb_mc_coordinate_get_y(tiles[tile_id]));
+			return error;
+		}
+	}
+
+	return HB_MC_SUCCESS; 
+}
+					
+					
+
+ 
+
+/*!
+ * Sets a Vanilla Core Endpoint's tile group's origin symbols __bsg_grp_org_x/y.
+ * Behavior is undefined if #mc is not initialized with hb_mc_manycore_init().
+ * @param[in] mc         A manycore instance initialized with hb_mc_manycore_init().
+ * @param[in] map        Eva to npa mapping. 
+ * @param[in] bin        Binary elf file.
+ * @param[in] bin_size   Size of binary file. 
+ * @param[in] coord      Tile coordinates to set the origin of.
+ * @param[in] origin     Origin coordinates.
+ * @return HB_MC_SUCCESS on success and HB_MC_FAIL on failure.
+ */
+int hb_mc_tile_set_origin_symbols (	hb_mc_manycore_t *mc,
+					hb_mc_eva_map_t *map,
+					unsigned char* bin,
+					size_t bin_size,
+					const hb_mc_coordinate_t *coord,
+					const hb_mc_coordinate_t *origin){
+
+	int error;
+	const hb_mc_config_t *cfg = hb_mc_manycore_get_config (mc); 
+
+	hb_mc_eva_t org_x_eva, org_y_eva;
+	error = hb_mc_loader_symbol_to_eva(bin, bin_size, "__bsg_grp_org_x", &org_x_eva); 
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_grp_org_x eva.\n", __func__);
```

Comment:
that's spelled "acquire"

Reply:
C needs a spell checker for strings; I mean really it's 2019!

Reply:
Ha yeah agreed! Thanks for pointing it out, I'll take care of it!